### PR TITLE
[DOCS-5403] adds DevCycle docs to RUM feature tracking

### PR DESCRIPTION
**Pull request by nikolasleblanc**
### What does this PR do?

Adds DevCycle specific documentation to the RUM Feature Flag tracking documentation, explaining how to send flag evaluation data to RUM when using DevCycle.

### Motivation

I saw that DataDog had a Feature Flag tracking system in Beta and as an employee of DevCycle (and client of DataDog's), I wanted to demonstrate how to send DevCycle variable evaluations to DataDog.

### Additional Notes

Some of my prettier/eslint autoformatting was applied to the partials file. Happy to remove that if it's problematic.

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by maycmlee** [comment]
Thanks @nikolasleblanc for opening this PR! We will review it and be back in touch.

**Comment by nikolasleblanc** [comment]
Thank you @maycmlee!

**Comment by VicV** [comment]
Hey @maycmlee ! 

Don't want to be a bother, but we've had our own clients come to us asking when we're going to be supported. 

We already have our own docs up: https://docs.devcycle.com/tools-and-integrations/

But our clients don't realize that we support this at DevCycle already because their entrypoint to knowing about this new RUM FF tracking is through the DD side.

**Comment by maycmlee** [comment]
Hi @VicV, thanks for the additional context! We'll get this reviewed soon!

**Review by bcaudan — APPROVED** [review]

**Comment by nikolasleblanc** [comment]
@maycmlee and @bcaudan thanks for the approval. I'm assuming I still need a few of the codeowner approvals before this can go in but keep me posted and I'll get this up to date.

File: content/en/real_user_monitoring/guide/setup-feature-flag-data-collection.md
Comment (rtrieu) [inline content/en/real_user_monitoring/guide/setup-feature-flag-data-collection.md:342]:
```suggestion
DevCycle does not support this integration. Create a ticket with DevCycle to request this feature.
```

File: content/en/real_user_monitoring/guide/setup-feature-flag-data-collection.md
Comment (rtrieu) [inline content/en/real_user_monitoring/guide/setup-feature-flag-data-collection.md:348]:
```suggestion
DevCycle does not support this integration. Create a ticket with DevCycle to request this feature.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Hi @nikolasleblanc, thanks so much for creating this PR! I just made a couple of suggestions based on our own style guide, but otherwise, it looks good and has been approved by our eng team. Once you make the changes, I will approve and merge it. Thanks again!

**Comment by rtrieu** [comment]
Hi @nikolasleblanc, I have a request for you. Would you mind creating an integration tile for this? [Here is some documentation](https://docs.datadoghq.com/developers/integrations/api_integration/) on what that is and how to get started. Please **ignore the details** around **Oauth** and **API endpoints**, since the integration does not need them.

[Here is the Github link to the code](https://github.com/DataDog/integrations-extras) where you will add the tile to (and you can create a new folder for the tile).

**Comment by nikolasleblanc** [comment]
Hey @rtrieu I've submitted a PR to the integrations-extra repo: https://github.com/DataDog/integrations-extras/pull/1849

**Comment by rtrieu** [comment]
@nikolasleblanc Thank you so much! Approving and merging this.

**Review by rtrieu — APPROVED** [review]
Looks good!

**Comment by rtrieu** [comment]
Hi @nikolasleblanc thank you so much for the quick turnaround on that!

**Pull request merged**
